### PR TITLE
Feature/esckan-65 - Generate PDF from the summary view

### PR DESCRIPTION
**Pull request by D-GopalKrishna**
Issue Link  - [ESCKAN-65](https://metacell.atlassian.net/browse/ESCKAN-65)

Task description:
1. Add a pdf generating service - `generatePDFService` -  generates a pdf as the below format. 
[file (12).pdf](https://github.com/user-attachments/files/16978004/file.12.pdf)

alternative is also to add the text KnowledgeStatement before writing the knowledge statement for that CS. As below:
[file (11).pdf](https://github.com/user-attachments/files/16978009/file.11.pdf)

2. Edit the header to download PDF instead of csv. 

Let me know @ddelpiano if this font works, still working on to get the Asap Font to work with the library. thanks! 

[ESCKAN-65]: https://metacell.atlassian.net/browse/ESCKAN-65?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/pdfService.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        return [];
+      });
+    });
+    if (connectionDetail.length !== 0) {
+      return connectionDetail.join(', ');
+    }
+    emptyColumns.push(index);
+    return null;
+  });
+  // remove the columns with the null values - to make the pdf more readable.
```
Comment (D-GopalKrishna) [inline src/services/pdfService.ts:150]: we remove the empty columns, so it is more readable when there are a lot of columns. @ddelpiano can you confirm if this is okay?

**Comment by D-GopalKrishna** [comment]
Example of the connectivity matrix:

image

**Comment by D-GopalKrishna** [comment]
Demo file 

[file - 2024-09-18T154259.598.pdf](https://github.com/user-attachments/files/17042405/file.-.2024-09-18T154259.598.pdf)

File: src/components/connections/SummaryHeader.tsx [hunk 2]
```diff
@@ -48,18 +59,33 @@ const SummaryHeader = ({
     }
   };
 
-  const generateCSV = () => {
-    const blob = generateJourneyCsvService(
-      selectedConnectionSummary?.['connections'],
-      selectedConnectionSummary?.endOrgan?.name ?? '',
-      filters,
+  pdfMake.fonts = {
```
Comment (afonsobspinto) [inline src/components/connections/SummaryHeader.tsx:62]: nitpick: I would move this to the pdf service

File: src/components/connections/SummaryHeader.tsx [hunk 3]
```diff
@@ -11,9 +11,19 @@ import { ArrowRight } from '../icons';
 import Breadcrumbs from '@mui/material/Breadcrumbs';
 import { SummaryType, KsRecord } from '../common/Types';
 import { useDataContext } from '../../context/DataContext.ts';
-import { generateJourneyCsvService } from '../../services/csvService.ts';
+import { generatePDFService } from '../../services/pdfService.ts';
+import pdfMake from 'pdfmake/build/pdfmake';
+import pdfFonts from 'pdfmake/build/vfs_fonts';
+import { TDocumentDefinitions } from 'pdfmake/interfaces';
+import { AsapFontBold, AsapFontBoldItalic, AsapFontItalic, AsapFontRegular } from '../../theme/AsapFontBase64.ts'
```
Comment (afonsobspinto) [inline src/components/connections/SummaryHeader.tsx:18]: question: Is it possible to fix the github actions complains?

File: src/services/pdfService.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    });
+    if (connectionDetail.length !== 0) {
+      return Array.from(new Set(connectionDetail)).join(', ');
+    }
+    emptyColumns.push(index);
+    return null;
+  });
+  // remove the columns with the null values - to make the pdf more readable.
+  const filteredViasRow =
+    columns.length > 6 ? viasRow.filter((v) => v !== null) : viasRow;
```
Comment (afonsobspinto) [inline src/services/pdfService.ts:152]: suggestion: Could 6 be a variable?

**Review by afonsobspinto — CHANGES_REQUESTED** [review]

File: src/components/connections/SummaryHeader.tsx (hunk 2 above)
Comment (D-GopalKrishna) [reply src/components/connections/SummaryHeader.tsx:62]: I moved part of the docDefinition to the pdf service, but I believe the `fonts` and `window.pdfMake.vfs` are mainly component-level definitions. I would say it's better to keep them within the component.

**Review by ddelpiano — CHANGES_REQUESTED** [review]
left a comment to gopal, I would add the ids to the new data returned other than only the labels.

**Pull request merged**